### PR TITLE
Concise dynamic node ids and names

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
PART 1:
 - Dynamic node ids currently have additional context, namely
a shortened name of the dynamic task function
 - This causes a tremendous bloat and is completely dependent on the
name of the task
 - This PR standardizes the name and affixes a simple `d` in the
begining

PART 2:
Reduce node names, which are purely for decorative purposes

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Tracking Issue
fixes https://github.com/lyft/flyte/issues/1660

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by wild-endeavor** [comment]
This won't work in certain dynamic cases if users are on the old version of Admin right?  before the proper node to node relationship was done.  Should we call that out?
